src/fu-common.c

```diff
@@ -643,3 +643,31 @@ fu_common_read_uint32 (const guint8 *buf, FuEndianType endian)
 	}
 	return val_native;
 }
+
+/**
+ * fu_common_get_machine_hash:
+ * @salt: The salt, or %NULL for none
+ * @error: A #GError or %NULL
+ *
+ * Gets a salted hash of the /etc/machine-id contents. This can be used to
+ * identify a specific machine. It is not possible to recover the original
+ * machine-id from the machine-hash.
+ *
+ * Returns: the SHA256 machine hash, or %NULL if the machine-id is not present
+ **/
+gchar *
+fu_common_get_machine_hash (const gchar *salt, GError **error)
+{
+	g_autofree gchar *buf = NULL;
+	g_autoptr(GChecksum) csum = NULL;
+	gsize sz = 0;
+
+	/* this has to exist */
+	if (!g_file_get_contents ("/etc/machine-id", &buf, &sz, error))
+		return NULL;
```

Comment:
I think you need to also check that the file is non-empty.  https://www.freedesktop.org/software/systemd/man/machine-id.html indicates it should be empty in cloud or container images, and you don't want to end up having those use this stuff.

Reply:
If it's empty, is that an error?

Reply:
Well if you're returning NULL when machine ID isn't present I think you also want to return that if it's a result that is guaranteed non-unique.

Reply:
Agreed, fixed.